package mvp.pipeline;

import "vw/Plate/PlateGeoReferenceDesc.proto";
import "mvp/Pipeline/AlgorithmSettings.proto";

message JobDesc {
  message Input {
    repeated string images = 1;
    repeated string cameras = 2;
  }

  message Render {
    required int32 col = 1;
    required int32 row = 2;
    required int32 level = 3;
    required bool use_octave = 5;
  }

  message Output {
    required string result = 1;
    required string internal_result = 2;
    required vw.platefile.PlateGeoReferenceDesc plate_georef = 3;
  }

  required int32 id = 1;
  required Input input = 2;
  required Render render = 3;
  required Output output = 4;
  required AlgorithmSettings algorithm = 5;
}
